### PR TITLE
Add feedback Implementation

### LINKED ISSUE #331: FEATURE: Feedback Implementation
### Is there an existing issue raised for this?

- [X] I have searched the existing issues

### Describe the feature

users submitting the feedback should get stored in the database associated with them. Optional implementation if needed i.e. sending thank you emails to the users.

Screenshot 2024-10-24 135807

### Component

backend/controllers/feedbackContoller

### It should be implemented because

Good feature fir a website like this and helps in testing the website for future upgrades and cutomizations

### Would you like to work on this issue?

None

### Terms & Conditions

- [X] I agree to follow the Code of Conduct
- [X] I'm a GSSOC'24 Ext contributor

### PR DESCRIPTION
**Pull request by haseebzaki-07**
## What does this PR do?

This PR is raised again for the issue in PR #349 
### Required

- [X] Filled out the "How to test" section in this PR
- [X] Read Contributing guide
- [X] Self-reviewed my own code
- [X] Checked for warnings, there are none
- [X] Removed all `console.logs`
- [X] The changes don't cause any responsiveness issues

### CODE REVIEW

**Comment by AlfiyaSiddique** [comment]
@haseebzaki-07 again add a video of output and make sure no errors in frontend console and backend terminal

File: backend/Controllers/UserController.js [hunk 1]
```diff
@@ -114,124 +120,43 @@ const verifyUserByToken = async (req, res) => {
...
-    const id = decoded.userId;
-    const hashPassword = await bcrypt.hash(password, 10);
-    await User.findByIdAndUpdate({ _id: id }, { password: hashPassword });
-    return res
-      .status(200)
-      .json({ success: true, message: "Password reset succesfully" });
-  } catch (error) {
-    return res.status(400).json({ success: false, message: "Invalid token" });
-  }
-};
```
Comment (AlfiyaSiddique) [inline backend/Controllers/UserController.js:153]: @haseebzaki-07 why the above functions are deleted ?? They are important

File: backend/models/Comment.js [hunk 2]
```diff
@@ -1,12 +1,13 @@
+// models/Comment.js
 import mongoose from 'mongoose';
 
 const commentSchema = new mongoose.Schema({
-  recipe: { type: mongoose.Schema.Types.ObjectId, ref: 'Recipe', required: true }, // Link to the related recipe
-  username: { type: String, required: true },     // User who made the comment
+  recipe: { type: mongoose.Schema.Types.ObjectId, ref: 'Recipe', required: true },
+  user: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true }, // Link to the user who made the comment
+  username: { type: String, required: true }, 
   content: { type: String, required: true },
```
Comment (AlfiyaSiddique) [inline backend/models/Comment.js:8]: Make sure this model change does not break any other API

**Comment by AlfiyaSiddique** [comment]
@haseebzaki-07 some comments please fix them

File: backend/Controllers/UserController.js [hunk 3]
```diff
@@ -51,6 +54,7 @@ const getAllUserName = async (req, res) => {
     names.forEach((val) => nameArr.push(val.username));
     res.status(200).json({ usernames: nameArr, success: true });
   } catch (error) {
+    console.log(error);
     console.log(error);
     res.status(404).json({ success: false, message: "Internal server error" });
```
[the remark this answers is not in the document]
Comment (haseebzaki-07) [reply backend/Controllers/UserController.js]: removed

File: backend/Controllers/UserController.js (hunk 1 above)
Comment (haseebzaki-07) [reply backend/Controllers/UserController.js:152]: @AlfiyaSiddique brought them back

File: backend/models/Comment.js (hunk 2 above)
Comment (haseebzaki-07) [reply backend/models/Comment.js]: @AlfiyaSiddique this was wrongly commited, fixed them to match the upstream

**Comment by haseebzaki-07** [comment]
Screenshot 2024-10-29 115801
Screenshot 2024-10-29 120429

https://github.com/user-attachments/assets/71b1d51c-bca5-4eea-842e-88f73b3a14f1

hey @AlfiyaSiddique made all the changes and the code is not breaking in both the frontend and the backend

**Pull request merged**